public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -241,6 +241,20 @@ export function getLogQueryFromMetricsQuery(query: string): string {
   return `${selector} ${pipelineExpr}`.trim();
 }
 
+export function getLogQueryFromMetricsQueryAtPosition(query: string, position: number): string {
+  if (isLogsQuery(query)) {
+    return query;
+  }
+
+  const metricQuery = getNodesFromQuery(query, [MetricExpr])
```

Comment:
You could also use `getNodePositionsFromQuery` and later replace `query.substring(metricQuery.from, metricQuery.to)` with `metricQuery.getExpression(query)`.

Reply:
Thought about that but eventually decided to go with the old route. Maybe it would be worth to have a brief discussion if we want to make this the default abstraction for nodes and refactor all the query functions, which should be really simple. For example, as a first step I would move the class to its own module to allow it to organically grow, independent of queryUtils.